src/arb_types.jl

```diff
@@ -83,6 +83,17 @@ mutable struct acb_struct
     end
 end
 
+mutable struct acb_vec_struct
+    entries::Ptr{acb_struct}
+    n::Int
+
```

Comment:
Should this get the same memory layout as `acb_mat`? There are functions like
```c
int acb_mat_solve(acb_mat_t X, const acb_mat_t A, const acb_mat_t B, slong prec)
```
which expect a matrix whereas in the Julia mindset a vector should also work. If `acb_vec_struct` has the same memory layout then this is all fine.

Reply:
Or maybe we don't need `acb_vec_struct` in the first place, just a "vector" construct for `acb_mat`?

Reply:
I don't quite understand this. Could you elaborate more? Do you mean that we should represent it as
```julia
mutable struct acb_vec_struct
    entries::Ptr{acb_struct}
    r::Int # one of these initialized to 1
    c::Int 
    rows::Ptr{Ptr{acb_struct}} # pointer to entries
    ...
end
```
?

Reply:
Yeah. I am just thinking about the confusion if we have a dedicated vector type, but which you cannot pass to solving a linear system.

Reply:
ok, I guess it should be then a row-major vector? i.e. (1 column)

Reply:
I think so

Reply:
I would say the solution to this is to write a wrapper for automatic conversion from `AcbVector` to an `AcbMatrix` of the correct shape. There are a lot of similar things about expectations in Julia that we have to write wrappers for I believe.